Tools/YamlCreate.ps1

```diff
@@ -220,59 +297,54 @@ Function Read-WinGet-InstallerValues {
         "UpgradeBehavior"
         "AnotherInstaller"
     )
-    Foreach ($InstallerValue in $InstallerValues) {Clear-Variable -Name $InstallerValue -Force -ErrorAction SilentlyContinue}
+    Foreach ($InstallerValue in $InstallerValues) { Clear-Variable -Name $InstallerValue -Force -ErrorAction SilentlyContinue }
 
-    while ([string]::IsNullOrWhiteSpace($InstallerUrl)) {
+    while (!(String.IsValid $InstallerUrl -MinLength 1 -MaxLength $Patterns.InstallerUrlMaxLength -MatchPattern $Patterns.InstallerUrl)) {
         Write-Host
         Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the download url to the installer.'
         $InstallerUrl = Read-Host -Prompt 'Url' | TrimString
     }
 
-    Write-Host
-    Write-Host -ForegroundColor 'White' "Save to disk?"
-    Write-Host "Do you want to save the files to the Temp folder?"
-    Write-Host -ForegroundColor 'White' -NoNewline "[Y] Yes  "
-    Write-Host -ForegroundColor 'Yellow' -NoNewline '[N] No  '
-    Write-Host -ForegroundColor 'White' -NoNewline "[M] Manually Enter SHA256 "
-    Write-Host -NoNewline "(default is 'N'): "
-    do {
-        $keyInfo = [Console]::ReadKey($false)
-    } until ($keyInfo.Key)
+    $_menu = @{
+        entries       = @("[Y] Yes"; "*[N] No"; "[M] Manually Enter SHA256")
+        Prompt        = "Do you want to save the files to the Temp folder?"
+        DefaultString = "N"
+    }
 
-    switch ($keyInfo.Key) {
-        'Y' {$SaveOption = '0'}
-        'N' {$SaveOption = '1'}
-        'M' {$SaveOption = '2'}
-        default {$SaveOption = '1'}
+    switch ( KeypressMenu -Prompt $_menu["Prompt"] -Entries $_menu["Entries"] -DefaultString $_menu["DefaultString"]) {
+        'Y' { $script:SaveOption = '0' }
+        'N' { $script:SaveOption = '1' }
+        'M' { $script:SaveOption = '2' }
+        default { $script:SaveOption = '1' }
     }
 
-    if ($SaveOption -ne '2') {
+    if ($script:SaveOption -ne '2') {
         Write-Host
         $start_time = Get-Date
         Write-Host $NewLine
         Write-Host 'Downloading URL. This will take a while...' -ForegroundColor Blue
         $WebClient = New-Object System.Net.WebClient
         $Filename = [System.IO.Path]::GetFileName($InstallerUrl)
-        $dest = "$env:TEMP\$FileName"
+        $script:dest = "$env:TEMP\$FileName"
 
         try {
-            $WebClient.DownloadFile($InstallerUrl, $dest)
+            $WebClient.DownloadFile($InstallerUrl, $script:dest)
         }
         catch {
             Write-Host 'Error downloading file. Please run the script again.' -ForegroundColor Red
             exit 1
         }
```

Comment:
This part seems to not be working and have for a good while, also we might want to ask for a new URL instead of exiting

Reply:
> This part seems to not be working and have for a good while, also we might want to ask for a new URL instead of exiting
> It does work locally, but not in the script..
As in that part hasn't worked at all in a while? What behavior is it exhibiting? Any way I can reproduce it?

Reply:
```
[Required] Enter the Package Identifier, in the following format <Publisher shortname.Application shortname>. For example: Microsoft.Excel
PackageIdentifier: Element.Element

[Required] Enter the version. for example: 1.33.7
Version: 1.33.7
Found Existing Version: 1.7.34

[Required] Enter the download url to the installer.
Url: https://packages.riot.im/desktop/install/win32/x64/Element%20Setup%201.33.7.exe 


Do you want to save the files to the Temp folder?
  [Y] Yes
  [N] No
  [M] Manually Enter SHA256

Enter Choice (default is 'N'): 

Downloading URL. This will take a while...
Error downloading file. Please run the script again.
Time taken: 0 second(s)
Resolve-Path : Cannot find path 'C:\Users\Esco\AppData\Local\Temp\Element%20Setup%201.33.7.exe' because it does not exist.
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psm1:110 char:36    
+                 $pathsToProcess += Resolve-Path $Path | Foreach-Objec ...
+                                    ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Esco\...up%201.33.7.exe:String) [Resolve-Path], ItemNotFoundException      
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.ResolvePathCommand
 
Get-AppLockerFileInformation : Cannot find path 'C:\Users\Esco\AppData\Local\Temp\Element%20Setup%201.33.7.exe' because it does 
not exist.
At C:\Users\Esco\GitHub\winget-pkgs\Tools\YamlCreate.ps1:340 char:32
+ ... Information = Get-AppLockerFileInformation -Path $script:dest | Selec ...
+                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (C:\Users\Esco\...up%201.33.7.exe:String) [Get-AppLockerFileInformation], ItemNotF  
   oundException
    + FullyQualifiedErrorId : CmdletHelper-FailedResolvingPSPath,Microsoft.Security.ApplicationId.PolicyManagement.Cmdlets.GetAppL  
   ockerFileInformationCmdlet
 
Remove-Item : Cannot find path 'C:\Users\Esco\AppData\Local\Temp\Element%20Setup%201.33.7.exe' because it does not exist.
At C:\Users\Esco\GitHub\winget-pkgs\Tools\YamlCreate.ps1:342 char:314
+ ... le', 'CurrentCultureIgnoreCase'))) { Remove-Item -Path $script:dest }
+                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Esco\...up%201.33.7.exe:String) [Remove-Item], ItemNotFoundException       
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand
```

Hold on, it does seem to work, just not exit? eitherway i think this might need a rework, to not fully exit the script, something like the PR Url check?